src/Suggestor/Suggestor.ts

```diff
@@ -0,0 +1,379 @@
+import type { Settings } from '../config/Settings';
+import { DateParser } from '../Query/DateParser';
+import { doAutocomplete } from '../DateAbbreviations';
+import { Recurrence } from '../Recurrence';
+
+import * as task from '../Task';
+
+const datePrefixCharacters = `${task.startDateSymbol}${task.scheduledDateSymbol}${task.dueDateSymbol}`;
+const maxTotalSuggestions = 6;
```

Comment:
In my copy, I increased this a lot. I thought it was better to see more suggestions and experiment and find that I could filter the list, rather than not be aware that there were other options I could search for.

Reply:
Do you think this should be configurable?
I'm personally not in favor of too many configuration options, but maybe this max value is too much of a personal taste.
A higher value has its benefits like you mention, but a smaller value (and thus a smaller-sized menu) is less intrusive when typing, and much more feasible for mobile.

Reply:
I think making it configurable would be good then. Though having 3 or 4 configurations for this one feature and having to restart after making every change, to test it, might get a bit old.